### PR TITLE
-s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=1 + --proxy-to-worker

### LINKED ISSUE #8733: EMSCRIPTEN_EVENT_TARGET_* throws with --proxy-to-worker
The old `#document`/`#window` still work fine. Can be reproduced by running `test_sdl2_canvas_proxy` with SDL2 master. Looks like `_specialEventTargets` is set before the proxied window/document objects exist.

### PR DESCRIPTION
**Pull request by juj**
Make -s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=1 to work with --proxy-to-worker mode.

Should fix #8733, although I was unable to test in exact conditions.

### CODE REVIEW

**Comment by Daft-Freak** [comment]
Fixes SDL2 with `DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=1`, but not with `=0` (It always uses the new constants). Works with the condition removed though.

**Comment by juj** [comment]
> ... Works with the condition removed though.

Sorry, can you rephrase? Do you mean that
 - `-s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=1` works, but
 - `-s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=0` does not, while
 - not specifying `DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=0/1` at all also works?

The option `DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR` defaults to value `0`, so not specifying it should be the same as specifying `-s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=0` - so I'm surprised if these two have different behavior?

**Comment by Daft-Freak** [comment]
Sorry, what I was trying to say was that both modes work with the `#if DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR` in this PR removed.

**Comment by juj** [comment]
Ohh, gotcha! Now I see what you mean. Indeed `_findEventTarget` references to `__specialEventTargets` also in `-s DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR=0` mode, so need to initialize that in both cases. Updated this PR, how does it look now?

**Comment by juj** [comment]
Test failures look intermittent.

**Comment by Daft-Freak** [comment]
Looks good, I'm not getting any failures on SDL2 master now.

**Comment by kripken** [comment]
Merging to allow the SDL2 update to happen sooner.
